packages/browser-data-collector/src/collectors/page-visibility.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * This set of reporters try to detect if the tab/window was 'hidden' at any point during the report duration.
+ * It doesn't care about the current state or when it becomes 'visible', that's why we don't listen for that
+ * state change.
+ *
+ * It is not perfect because it can't detect if the tab/window was hidden between the page laod and when the
+ * start collector is called.
+ *
+ * It also assumes there is only one report running. If we ever have more than one, we may have problems when
+ * unsubscribing from the `visibilitychange` event.
+ */
```

Comment:
This approach seems quite reasonable. It's not perfect, as you say, but as far as I can tell very few user journeys will result in it reporting the wrong thing. And it's not critical data, in any case; it may introduce a bit of confusion into our analysis, but it won't break anything.